ehrapy/anndata/anndata_ext.py

```diff
@@ -106,7 +105,6 @@ def df_to_anndata(
         X=X,
         obs=_cast_obs_columns(dataframes.obs),
         var=pd.DataFrame(index=list(dataframes.df.columns)),
-        dtype="float32" if all_num else "object",
```

Comment:
Does this work well with all numeric anndata objects? Or is this patched in latest anndata version?

Reply:
I now have 
```

    all_num = True if len(numerical_columns) == len(list(dataframes.df.columns)) else False
    X = X.astype(np.number) if all_num else X.astype(object)
    # cast non numerical obs only columns to category or bool dtype, which is needed for writing to .h5ad files
    adata = AnnData(
        X=X,
        obs=_cast_obs_columns(dataframes.obs),
        var=pd.DataFrame(index=list(dataframes.df.columns)),
        layers={"original": X.copy()},
        uns=uns,
    )
```

which should solve both of your comments